### ISSUE TITLE
🏗️ Tech Debt: Deprecated transformers imports and copied generation code

### ISSUE THREAD

**Issue by vietanhrs** [dinhanhx/VisualRoBERTa#4]
### 1. Deprecated `transformers` imports (`src/vision_language.py`, lines 21-35)
```python
from transformers.generation_logits_process import LogitsProcessorList
from transformers.generation_stopping_criteria import ...
from transformers.generation_utils import ...
from transformers.generation_beam_search import ...
from transformers.pytorch_utils import torch_int_div
```

These modules were restructured in `transformers >= 4.30`. The current imports will break with newer versions:
- `transformers.generation_logits_process` → `transformers.generation.logits_process`
- `transformers.generation_utils` → `transformers.generation.utils`
- `torch_int_div` was removed entirely (use `//` operator)

### 2. Hundreds of lines of copied generation code
`greedy_search()` and `beam_search()` are copy-pasted from transformers with minimal modifications (only the `next_token_logits` line differs). This duplicated code will diverge from upstream fixes.

### 3. Unreachable code after `raise NotImplementedError`
Methods `sample()`, `beam_sample()`, `group_beam_search()`, and `constrained_beam_search()` all have `return super()...` after `raise NotImplementedError` — dead code.

## Why It Matters
The code is pinned to an old transformers version and will break on upgrade. The copied generation methods are a maintenance burden.

## Suggested Fix
Override only the logits extraction by modifying `prepare_inputs_for_generation` and using a logits processor instead of copying entire generation methods.
